magic/fetcher.py

```diff
@@ -3,14 +3,70 @@
 import os
 from collections import OrderedDict
 from urllib import parse
+from functools import wraps
+import time as py_time
 
 import pkg_resources
 from github import Github
 
 import magic.fetcher_internal as internal
 from magic.fetcher_internal import FetchException
+from magic import oracle #for card-by-name for scryfall
 from shared import configuration, dtutil
 
+
+def stagger(delay=0.1):
+    def decorator(func_in):
+        @wraps(func_in)
+        def func_out(*args, **kwargs):
+            if py_time.time() - func_out.last_call < delay:
+                func_out.last_call = py_time.time() + delay
+                py_time.sleep(delay - (py_time.time() - func_out.last_call))
+            return func_in(*args, **kwargs)
+        func_out.last_call = float("-inf")
+        return func_out
+    return decorator
+
+
+@stagger(0.1)
```

Comment:
I don't think it's ever actually staggered anything. Scryfall requests that we wait 0.1s between requests, but there's no way we're ever realistically doing that anyway.